### DISCUSSION TITLE
Working with permissions

### DISCUSSION THREAD

**Discussion by hanno00** [capacitor-community/bluetooth-le#501] [General]
### Permission handling

Hi,

I've been having some issues regarding the Bluetooth permissions while using this plugin.
I want to prompt a user with a popup before calling `initialize()`, so that I can prepare the users before showing the permission dialog.

For this functionality I wanted to use the `checkPermissions()` and `requestPermissions()` methods, defined in the [capacitor documentation](https://capacitorjs.com/docs/plugins/android#permissions).

However, I am unable to use these methods. I don't know if I am not doing it correctly or if they aren't implemented in this plugin.

I would highly appreciate any information regarding this subject.

Thanks in advance,

Hanno

| Tool            | Version              |
|-----------------|----------------------|
| Android         | 12                   |
| Ionic CLI       | 6.20.8               |
| Ionic Framework | @ionic/angular 6.5.5 |
| Capacitor CLI   | 4.6.3                |
| NodeJs          | v14.20.0             |
| OS              | Windows 10           |

**Comment by pwespi** [comment]
The `checkPermissions` and `requestPermissions` methods are currently not implemented by this plugin.

See feature request https://github.com/capacitor-community/bluetooth-le/issues/487

**Reply by hanno00** [reply]
Okay thank you :)
